modular_zzplurt/readme.md

```diff
@@ -0,0 +1,165 @@
+
+If you'd like to know more about coding, contributing and contribution standards, feel free to read this repository's [contribution guides](./.github/guides)!
+
+## Important Note - Test Your Pull Requests
+
+You are responsible for testing your content. Please do not mark a pull request as ready for review until you have thoroughly tested it. If you need a separate client for testing, you can use a guest account by logging out of BYOND and connecting to your test server. Test merges are for stress tests, not for finding bugs that could have been caught with local testing.
+
+## The Nature of Conflicts
+
+Conflicts can arise when changes are made to the same lines of code in different branches. For example, if the original code is:
+
+```byond
+var/something = 1
+```
+
+And we change it to:
+
+```diff
+- var/something = 1
++ var/something = 2 //SPLURT EDIT
+```
+
+But upstream changes it to:
+
+```diff
+- var/something = 1
++ var/something = 4
+```
+
+This results in a conflict that needs to be resolved manually. Our solution is modularization.
+
+## The Modularization Protocol
+
+Modularization is the practice of organizing code into separate, self-contained modules that can be developed, tested, and maintained independently. In the context of our codebase, this means placing as much as possible of the new code, icons, sounds, and other assets into the `modular_zzplurt` folder. This approach helps keep our core codebase clean and reduces the likelihood of conflicts when merging changes. By following a structure that resembles the main repository within the `modular_zzplurt` folder, we ensure that our modular code is easy to navigate and manage. This also allows us to make changes and add new features without directly modifying the core files, thereby maintaining the integrity and stability of the main codebase.
+
+Our modularization protocols are founded on three pillars:
+
+1. **Modularize anything that's reasonable to modularize:** This includes anything that can be modularized without decreasing its quality or the quality of existing code, and won't logically cause more issues by modularizing than not doing it.
+2. **Use commenting conventions for non-modular edits:** When editing non-modular code, make sure to adequately use the commenting conventions to ensure it is known that it's a non-modular edit.
+3. **It's ok to mess up:** These rules are intended less as a set of guidelines for contributors to strictly follow, but rather coding standards that we all collectively help to attain, for examble through maintainers helping any contributors fulfill them in their pull requests.
```

Comment:
```suggestion
3. **It's ok to mess up:** These guidelines are intended less as a set of ironclad rules for contributors to strictly follow, but rather coding standards that we all collectively help to attain, for example through maintainers helping any contributors fulfill them in their pull requests.
```
I would call these guidelines rather than rules, personally, but that's again a matter of personal taste in writing. Also, typo; `examble` should be `example`.